homatransport/src/transport/VectioTransport.h

```diff
@@ -0,0 +1,142 @@
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see http://www.gnu.org/licenses/.
+//
+
+#ifndef __HOMATRANSPORT_VECTIOTRANSPORT_H_
+#define __HOMATRANSPORT_VECTIOTRANSPORT_H_
+
+#include <omnetpp.h>
+#include <unordered_map>
+#include <list>
+#include "common/Minimal.h"
+#include "inet/transportlayer/contract/udp/UDPSocket.h"
+#include "application/AppMessage_m.h"
+#include "transport/HomaPkt.h"
+
+/**
+ * An near optimal priority based transport scheme for minimizing the average
+ * completion time of messages. For every packet of a message, it sets the
+ * message size field and a priority field equal to the remaining bytes of the
+ * message not yet send. This transport will only be near optimal if the network
+ * has priority queues and only for many-senders/single-receiver scenario. The
+ * scheduling mechanism for the priority queues in the network would always
+ * choose the lowest priority packet that belongs to shortes message among all
+ * packets in the queue.
+ */
+
+class VectioTransport : public cSimpleModule
+{
+  PUBLIC:
```

Comment:
Let's replace these capital case PUBLIC/PRIVATE/etc. with lower cases. I originally had them for testing purpose.